grr/server/grr_response_server/gui/ui/components/client_details/client_details.ng.html

```diff
@@ -0,0 +1,241 @@
+<div *ngIf="client$ | async as client" class="info">
+  <div class="header mat-body-1">
+    <mat-icon>computer</mat-icon>
+    <div class="mat-h1">{{client.knowledgeBase.fqdn}}
+      <div class="client-id mat-h3">{{client.clientId}}</div>
+    </div>
+  </div>
+  <div>
+    <div class="section-title mat-h3">General</div>
+    <table class="section">
+      <tr>
+        <td class="description">Client ID</td>
+        <td class="value">{{client.clientId}}</td>
+      </tr>
+      <tr *ngIf="client.knowledgeBase.fqdn">
+        <td class="description">FQDN</td>
+        <td class="value">{{client.knowledgeBase.fqdn}}</td>
+      </tr>
+      <tr *ngIf="client.lastSeenAt">
+        <td class="description">Last seen</td>
+        <td class="value"><timestamp [date]="client.lastSeenAt"></timestamp></td>
+      </tr>
+      <tr *ngIf="client.lastClock">
+        <td class="description">Last clock</td>
+        <td class="value"><timestamp [date]="client.lastClock"></timestamp></td>
+      </tr>
+      <tr *ngIf="client.lastBootedAt">
+        <td class="description">Last boot</td>
+        <td class="value"><timestamp [date]="client.lastBootedAt"></timestamp></td>
+      </tr>
+      <tr>
+        <td class="description">Labels</td>
+        <td class="value">
+          <mat-chip-list>
+            <mat-chip *ngFor="let label of client.labels" class="chip">
+              {{label.name}}
+            </mat-chip>
+          </mat-chip-list>
+        </td>
+      </tr>
+    </table>
+
+    <br>
+    <div class="section-title mat-h3">OS Info</div>
+    <table class="section">
+      <tr *ngIf="client.osInfo.system">
+        <td class="description">System</td>
+        <td class="value">{{client.osInfo.system}}</td>
+      </tr>
+      <tr *ngIf="client.osInfo.release">
+        <td class="description">Release</td>
+        <td class="value">{{client.osInfo.release}}</td>
+      </tr>
+      <tr *ngIf="client.osInfo.version">
+        <td class="description">Version</td>
+        <td class="value">{{client.osInfo.version}}</td>
+      </tr>
+      <tr *ngIf="client.osInfo.kernel">
+        <td class="description">Kernel</td>
+        <td class="value">{{client.osInfo.kernel}}</td>
+      </tr>
+      <tr *ngIf="client.knowledgeBase.osMajorVersion">
+        <td class="description">OS major version</td>
+        <td class="value">{{client.knowledgeBase.osMajorVersion}}</td>
+      </tr>
+      <tr *ngIf="client.knowledgeBase.osMinorVersion">
+        <td class="description">OS minor version</td>
+        <td class="value">{{client.knowledgeBase.osMinorVersion}}</td>
+      </tr>
+      <tr *ngIf="client.osInfo.installDate">
+        <td class="description">Install date</td>
+        <td class="value"><timestamp [date]="client.osInfo.installDate"></timestamp></td>
+      </tr>
+    </table>
+
+    <br>
+    <div class="section-title mat-h3">Users</div>
+    <table class="section">
+      <ng-template ngFor let-user [ngForOf]="client.users" let-i="index">
```

Comment:
What's the benefit of this syntax over `*ngFor="let user of client.users; index as i"`?

Reply:
This is the syntax when using `ng-template`, according to the docs: https://angular.io/api/common/NgForOf#description